api/types.proto

```diff
@@ -21,12 +21,22 @@ message Annotations {
 	map<string, string> labels = 2;
 }
 
-message Resources {
-	// Amount of CPUs (e.g. 2000000000 = 2 CPU cores)
-	int64 nano_cpus = 1 [(gogoproto.customname) = "NanoCPUs"];
+enum PredefinedResources {
```

Comment:
In particular, this PR misses the goal in that we have two classes of resources, "predefined" and some other class that is not as well supported. We need to have a model that represents them both equally.